src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -39,16 +54,20 @@ void uct_rdmacm_cm_ep_error_cb(uct_rdmacm_cm_ep_t *cep,
     }
 
     ucs_assert(status != UCS_OK);
-    cep->flags |= UCT_RDMACM_CM_EP_FAILED;
+    /* Do not set local EP failed on normal disconnect event */
+    if (status != UCS_ERR_CONNECTION_RESET) {
+        cep->flags |= UCT_RDMACM_CM_EP_FAILED;
+    }
```

Comment:
wrap this to another function - uct_rdmacm_cm_ep_set_ep_failed() which sets failed flag